kaos/algorithm/visibility_finder.py

```diff
@@ -227,13 +254,10 @@ def find_visibility(self, time_interval):
             calculated.
 
         """
-        # Calculate angle of visibility theta.
-        roots = mp.polyroots(self.find_approx_coeffs(time_interval), maxsteps=2000,
-                             extraprec=110)
-
+        roots = Solver.solve(*self.find_approx_coeffs(time_interval))
         return roots
 
-    def determine_visibility(self, error=0.1, tolerance_ratio=0.1, max_iter=100):
+    def determine_visibility(self, error=0.001, tolerance_ratio=0.1, max_iter=100):
```

Comment:
NOTE: We should probably come up for a reason behind those numbers when we submit to the client

Reply:
Anything above 0.001 misses entire visibility intervals.